hathor/indexes/utxo_index.py

```diff
@@ -0,0 +1,326 @@
+                continue
+            if heightlock is None:
+                timelock = address_script.get_timelock()
+            else:
+                # XXX: timelock forced to None when there is a heightlock
+                timelock = None
+            # XXX: that is, at least one of them must but None
+            assert timelock is None or heightlock is None
+            self._remove_utxo(UtxoIndexItem(
+                token_uid=tx.get_token_uid(tx_output.get_token_index()),
+                tx_id=tx.hash,
+                index=index,
+                address=address_script.address,
+                amount=tx_output.value,
+                timelock=timelock,
+                heightlock=heightlock,
+            ))
+        # the spent inputs have now become unspent, re-add them to the index
+        for tx_input in tx.inputs:
+            spent_tx = tx.get_spent_tx(tx_input)
+            spent_tx_meta = spent_tx.get_metadata()
+            spent_tx_output = spent_tx.outputs[tx_input.index]
+            address_script = parse_address_script(spent_tx_output.script)
+            if not address_script:
+                # XXX: completely ignore scripts that aren't "addressable"
+                continue
+            if spent_tx.is_block:
+                # XXX: timelock forced to None when there is a heightlock
+                timelock = None
+                heightlock = spent_tx_meta.height + settings.REWARD_SPEND_MIN_BLOCKS
+            else:
+                timelock = address_script.get_timelock()
+                heightlock = None
+            assert timelock is None or heightlock is None
+            self._add_utxo(UtxoIndexItem(
+                token_uid=spent_tx.get_token_uid(spent_tx_output.get_token_index()),
+                tx_id=tx_input.tx_id,
+                index=tx_input.index,
+                address=address_script.address,
+                amount=spent_tx_output.value,
+                timelock=timelock,
+                heightlock=heightlock,
+            ))
+        self._remove_tx(tx.hash)
+
+    def iter_utxos(self, *, token_uid: bytes, address: str, target_amount: int, target_timestamp: Optional[int] = None,
+                   target_height: Optional[int] = None) -> Iterator[UtxoIndexItem]:
+        iter_nolock = self._iter_utxos_nolock(token_uid=token_uid, address=address,
+                                              target_amount=target_amount)
+        iter_timelock = self._iter_utxos_timelock(token_uid=token_uid, address=address,
+                                                  target_amount=target_amount,
+                                                  target_timestamp=target_timestamp)
+        iter_heightlock = self._iter_utxos_heightlock(token_uid=token_uid, address=address,
+                                                      target_amount=target_amount,
+                                                      target_height=target_height)
+        iter_utxos = peekable(sorted_merger(
```

Comment:
Why `peekable(...)`? I can't see where it's in use here.

Reply:
You're right, it's not in use. I think I was using it in an early implementation but ended up doing it without it.

Reply:
Removed.

Reply:
Thanks!